src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useContext, useMemo } from 'react';
+
+import { TodoInfo } from '../TodoInfo';
+import { AppContext } from '../AppContext';
+import { FilterMode } from '../../types/FilterMode';
+
+export const TodoList: React.FC = React.memo(() => {
+  const {
+    allTodos,
+    tempTodo,
+    currentFilterMode,
+  } = useContext(AppContext);
+
+  const visibleTodos = useMemo(() => (
+    allTodos.filter(({ completed }) => {
+      switch (currentFilterMode) {
+        case FilterMode.Active:
+          return !completed;
+
+        case FilterMode.Completed:
+          return completed;
+
+        default:
+          return true;
+      }
+    })
+  ), [allTodos, currentFilterMode]);
+
+  window.console.log('Rendering todo list');
```

Comment:
Remove the `window.console.log` statement for better code quality and to avoid unnecessary logs in the production environment. 🧹